packages/core-flows/src/definition/cart/steps/link-cart-region.ts

```diff
@@ -0,0 +1,36 @@
+import { Modules } from "@medusajs/modules-sdk"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+export const linkCartRegionStepId = "link-cart-region"
+export const linkCartRegionStep = createStep(
+  linkCartRegionStepId,
+  async (data: { regionId: string; cartId: string }, { container }) => {
```

Comment:
Let's extract the type and use it below as well, so we don't have to do `data!`

Reply:
The type of data in the rollback function is inferred by the second argument of the step response, so here data will be `{ cartId, regionId } | undefined`. Now, we decided to make it also undefined, because it can happen that the rollback end up being called with no data in case the invoke function fail. I think here it is mostly an oversight, the first thing to do in that rollback is checking if that exists or return early, then data will not required `!`